cmd/juju/commands/bootstrap.go

```diff
@@ -843,15 +844,25 @@ See `[1:] + "`juju kill-controller`" + `.`)
 		}
 	}()
 
-	// Block interruption during bootstrap. Providers may also
-	// register for interrupt notification so they can exit early.
+	// Handle Ctrl-C during bootstrap by asking the bootstrap process to stop
+	// early (and the above will then clean up resources).
 	interrupted := make(chan os.Signal, 1)
 	defer close(interrupted)
 	ctx.InterruptNotify(interrupted)
 	defer ctx.StopInterruptNotify(interrupted)
+	cancelCtx, cancel := context.WithCancel(context.Background())
 	go func() {
 		for range interrupted {
-			ctx.Infof("Interrupt signalled: waiting for bootstrap to exit")
+			// Space prefix is intentional, so output appears as
+			// "^C Ctrl-C pressed" instead of "^CCtrl-C pressed".
+			_, _ = fmt.Fprintln(ctx.GetStderr(), " Ctrl-C pressed, stopping bootstrap and cleaning up resources")
```

Comment:
Either add a space or perhaps a newline would be better to keep the ^C separate from the Juju message

Reply:
Done as a newline in https://github.com/juju/juju/pull/12475